integration-cli/docker_cli_network_unix_test.go

```diff
@@ -1537,7 +1537,7 @@ func (s *DockerSuite) TestUserDefinedNetworkConnectDisconnectAlias(c *testing.T)
 	assert.NilError(c, err)
 
 	// verify the alias option is rejected when running on predefined network
-	out, _, err := dockerCmdWithError("run", "--rm", "--name=any", "--net-alias=any", "busybox:glibc", "top")
+	out, _, err := dockerCmdWithError("run", "--rm", "--name=any", "--net-alias=any", "busybox:glibc", "true")
```

Comment:
I have a suspicion what's causing it, but did a bit of digging at which point validation is performed, and it's (erm) horribly complex; validation is indeed _only_ done when _starting_ the container (which I think is wrong; at least _some_ of the validation should be done when _creating_ the container, so that we can fail early.

This is the code path taken;

1. [`daemon.ContainerStart()`](https://github.com/moby/moby/blob/de5a67156b5eed287883958b60b0c08dacfc7332/daemon/start.go#L21)
2. [`daemon.containerStart()`](https://github.com/moby/moby/blob/de5a67156b5eed287883958b60b0c08dacfc7332/daemon/start.go#L98)
3. [`daemon.initializeNetworking()`](https://github.com/moby/moby/blob/de5a67156b5eed287883958b60b0c08dacfc7332/daemon/start.go#L153)
4. [`daemon.allocateNetwork`](https://github.com/moby/moby/blob/de5a67156b5eed287883958b60b0c08dacfc7332/daemon/container_operations.go#L976)

Here it gets interesting, because there's a special case when no networks are defined; it will call [`daemon.updateContainerNetworkSettings(container, nil)`](https://github.com/moby/moby/blob/de5a67156b5eed287883958b60b0c08dacfc7332/daemon/container_operations.go#L538-L541), which does a lot of magic, and converts the options to a `networksettings.Network`; https://github.com/moby/moby/blob/de5a67156b5eed287883958b60b0c08dacfc7332/daemon/container_operations.go#L493-L500

After completing that, it will connect to the default network (if found in the `networksettings.Network` on the container); https://github.com/moby/moby/blob/de5a67156b5eed287883958b60b0c08dacfc7332/daemon/container_operations.go#L547-L550

Which will first find (and attach) the container to the network (??) (or return silently if the network was not found, perhaps for swarm-scoped networks?); https://github.com/moby/moby/blob/de5a67156b5eed287883958b60b0c08dacfc7332/daemon/container_operations.go#L736

Then, manipulate some of the settings again, and call [`daemon.updateNetworkConfig`](https://github.com/moby/moby/blob/de5a67156b5eed287883958b60b0c08dacfc7332/daemon/container_operations.go#L760), which (again) manipulates [some of the settings](https://github.com/moby/moby/blob/de5a67156b5eed287883958b60b0c08dacfc7332/daemon/container_operations.go#L684-L708)

And **_FINALLY_** will call [`validateNetworkingConfig()`](https://github.com/moby/moby/blob/de5a67156b5eed287883958b60b0c08dacfc7332/daemon/container_operations.go#L710), which will then validate if what we have is actually valid in `validateNetworkingConfig()`, where it will produce the error if the network is the default network, and if it has a custom IP or alias set; https://github.com/moby/moby/blob/cd19a333c8cf592e776eedefea69e6e62cbb1cfa/daemon/container_operations.go#L625-L630


... after which, it will do [many of these steps _again_ for the remaining (non-default) networks](https://github.com/moby/moby/blob/de5a67156b5eed287883958b60b0c08dacfc7332/daemon/container_operations.go#L563-L571)